src/backend/apidb/writeable_pgsql_selection.cpp

```diff
@@ -68,6 +71,27 @@ template <> struct string_traits<vector<tile_id_t> > {
     return ostr.str();
   }
 };
+
+// and again for osm_changeset_id_t
+template <> struct string_traits<vector<osm_changeset_id_t> > {
+  static const char *name() { return "vector<osm_changeset_id_t>"; }
+  static bool has_null() { return false; }
+  static bool is_null(const vector<osm_changeset_id_t> &) { return false; }
+  static stringstream null() {
+    internal::throw_null_conversion(name());
+    // No, dear compiler, we don't need a return here.
+    throw 0;
+  }
+  static void from_string(const char[], vector<osm_changeset_id_t> &) {}
+  static std::string to_string(const vector<osm_changeset_id_t> &ids) {
+    stringstream ostr;
+    ostr << "{";
+    std::copy(ids.begin(), ids.end(),
+              infix_ostream_iterator<osm_changeset_id_t>(ostr, ","));
+    ostr << "}";
+    return ostr.str();
+  }
+};
```

Comment:
Not as reasonable as I'd have liked (i.e: hiding it away in a `.cpp` file), but I've reduced it to a macro in fb6ad22 which at least means there's only one copy of the code in the human-readable sense.